internal/errors/option.go

```diff
@@ -23,4 +23,38 @@ var (
 		}
 		return Errorf("invalid option. name: %s, val: %#v", name, val)
 	}
```

Comment:
I think there should be `ErrInvalidOption` type not just using Errorf.
```go
type ErrInvalidOption struct {
	name string
	val interface{}
	err error
}
```
then, we can recognize which error is a ErrInvalidOption error using IsErrInvalidOption function.